### PR TITLE
refactor(libstatus): make use of ChatType enum

### PR DESCRIPTION
**Pull request by PascalPrecht**

### CODE REVIEW

File: config.nims [hunk 1]
```diff
@@ -1,6 +1,3 @@
-when defined(macosx):
-  import algorithm, strutils
-
```
Comment (PascalPrecht) [inline config.nims:3]: This one needs double check. I assume there's a reason those are conditionally imported. But still, nim complains they aren't used... so maybe someone can verify this?

**Comment by jrainville** [comment]
@PascalPrecht I just tested your PR on Ubuntu and opening a chat doesn't load the past messages and then sending a message crashes the app. It might be related to that `when defined(macosx):` line that you removed

**Comment by PascalPrecht** [comment]
@jrainville Can you try again?

> It might be related to that when defined(macosx): line that you removed

I could actually reproduce this but the issue was the usage of `ChatType` enum. Removed it for now

**Comment by jrainville** [comment]
@PascalPrecht Tested again and it worked!

**Pull request merged**